etc/linux.aarch64.mk

```diff
@@ -0,0 +1,45 @@
+# Get the toolchain paths for openmrn
+include $(OPENMRNPATH)/etc/path.mk
+
+
+ifndef TOOLPATH
+#TOOLPATHCOMMAND := $(shell \
+#sh -c "which aarch64-linux-gnu-gcc" \
+#)
+TOOLPATH := $(AARCH64LINUXGCCPATH)
+endif
+
+$(info armv7alinux toolpath '$(TOOLPATH)')
+
+# Get the $(CFLAGSENV), $(CXXFLAGSENV), $(LDFLAGSENV)
+include $(OPENMRNPATH)/etc/env.mk
+
+CC = $(TOOLPATH)/aarch64-linux-gnu-gcc
+CXX = $(TOOLPATH)/aarch64-linux-gnu-g++
+AR = $(TOOLPATH)/aarch64-linux-gnu-ar
+LD = $(TOOLPATH)/aarch64-linux-gnu-g++
+OBJDUMP = $(TOOLPATH)/aarch64-linux-gnu-objdump
+
+AROPTS=D
+
+HOST_TARGET := 1
```

Comment:
Is your intention to use this target *on* the device to compile stuff locally, or on your desktop machine to perform cross-compilation to the target, then copy the binaries?
A host target usually has /usr/bin/g++ as CXX. The paths above seem like a cross-compiling target.

Generally linux.x86 happens to work as an architecture-unspecific host target. It works for local compilation on all linux distributions and all architectures, including raspberry pis, beaglebones and others, also 32-bit and 64-bit. Yes, it is misnamed that way and should not really have x86 in the name of the target.

Reply:
Actually, like the linux.armv7a target, it is actually both:  Right now, I compile (using the local "native" compiler) on one Raspberry Pi binaries for other 32-bit ARM Linux machines (other Raspberry Pis or Beagleboards).  Or I could run the program locally on the build system as a workbench test. I see the same thing here.

While in the generic case using the "linux.x86" target for all arch. Linux builds works, the linux.armv7a and linux.aarch64 have a few arch. specific optimizations (-march=).  The paths are not actually cross-compiling targets.  I just copied what was already there for linux.armv7a.  I guess it could be a cross-compile as well as native.